source/Subsystem/EntityComponentSystem.h

```diff
@@ -1,13 +1,16 @@
 #pragma once
 
 #include "../Element/Entity.h"
+#include "../Component/TransformComponent.h"
 
 #include <string>
 
+class Engine;
+
 class EntityComponentSystem
 {
 public:
-	EntityComponentSystem() {};
+	EntityComponentSystem(Engine* aEngine) : myEngine(aEngine) {};
```

Comment:
We should use this forward declaring system for all subsystems so they all have a way to access the other subsystems when required.

Reply:
Maybe look into Service Locators/Provider pattern to decouple code like this.
e.g. https://gameprogrammingpatterns.com/service-locator.html
or Robert Nyholms' "Game Programming Patterns - Decoupling Pattern & Service Locator" chapter.

Reply:
Oooh, this is super interesting! I think I'll stick with the current pattern for now, where all subsystems have a reference back to the engine to be able to access the other subsystems, as we're only doing it for all classes that are tightly coupled to the engine. We can use the subsystems as a sort of interfacing layer with the engine. We could even consider making the engine class a singleton, but as we all know, singletons are evil.